internal/disk/customizations.go

```diff
@@ -0,0 +1,179 @@
+package disk
+
+import (
+	"io"
+	"math/rand"
+
+	"github.com/google/uuid"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/distro"
+)
+
+const (
+	sectorSize            = 512
+	BIOSBootPartitionGUID = "21686148-6449-6E6F-744E-656564454649"
+	BIOSBootPartitionUUID = "FAC7F1FB-3E8D-4137-A512-961DE09A5549"
+
+	FilesystemDataGUID = "0FC63DAF-8483-4772-8E79-3D69D8477DE4"
+	FilesystemDataUUID = "CB07C243-BC44-4717-853E-28852021225B"
+
+	EFISystemPartitionGUID = "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+	EFISystemPartitionUUID = "68B2905B-DF3E-4FB3-80FA-49D1E773AA33"
+	EFIFilesystemUUID      = "7B77-95E7"
+
+	RootPartitionUUID = "6264D520-3FB9-423F-8AB8-7A0A8E3D3562"
+)
+
+func CreatePartitionTable(
```

Comment:
The `disk` package should be IMO agnostic of existing image types and therefore its exported functions should not contain arguments and logic specific to an image type. My idea was that the provided `basePartitionTable` should contain the "default" partition table, including default partitioning layout, for the specific image type and architecture. This however means that the function may need to potentially recalculate some values - e.g. size of the root partition.